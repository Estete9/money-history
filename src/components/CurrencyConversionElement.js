import PropTypes from 'prop-types';
import styles from '../styles/currencyConversionElement.module.css';
import arrowRightImg from '../assets/arrow-right.svg';

function CurrencyConversionElement({ currency }) {
  return (
    <div className={styles.currencyConversionElementWrapper}>
      <div>{currency.currencySymbol.toUpperCase()}</div>
      <p className={styles.currencyConversion}>
        {`${currency.currencyCode} ${
          currency.value !== parseInt(currency.value, 10)
            ? currency.value.toFixed(4)
            : currency.value
        }`}
      </p>
      <img
        className={styles.conversionArrowImg}
        src={arrowRightImg}
        alt="enter currency conversion button"
      />
    </div>
  );
}

CurrencyConversionElement.propTypes = {
  currency: PropTypes.shape({
    currencySymbol: PropTypes.string.isRequired,
    value: PropTypes.number.isRequired,
    currencyCode: PropTypes.string.isRequired,
  }),
};

CurrencyConversionElement.defaultProps = {
  currency: {
    currencySymbol: 'No Symbol',
    value: 0,
    currencyCode: 'No CurrencyCode',
  },
};
export default CurrencyConversionElement;
