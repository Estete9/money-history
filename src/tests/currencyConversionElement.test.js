import React from 'react';
import renderer from 'react-test-renderer';
import CurrencyConversionElement from '../components/CurrencyConversionElement';

test('should render CurrencyConversionElement', () => {
  const currency = {
    year: '2023',
    currencySymbol: 'GBP',
    value: '1.65',
  };
  const component = renderer.create(<CurrencyConversionElement currency={currency} />);
  const conversionElementTree = component.toJSON();
  expect(conversionElementTree).toMatchSnapshot();
});
