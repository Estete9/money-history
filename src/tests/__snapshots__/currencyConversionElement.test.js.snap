// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CurrencyConversionElement 1`] = `
<div
  className="currencyConversionElementWrapper"
>
  <div>
    2023
  </div>
  <p
    className="currencyConversion"
  >
    GBP 1.65 = eur 1
  </p>
  <img
    alt="enter currency history button"
    className="historyArrowImg"
    src="svg-mock"
  />
</div>
`;
